examples/motoko_examples/persistent-storage/canisters/rust/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+use std::cell::RefCell;
+
+thread_local! {
+    static PERF_RESULT_REF_CELL: RefCell<Option<PerfResult>> = RefCell::default();
+}
+
+#[derive(candid::CandidType, Clone)]
+pub struct PerfResult {
+    wasm_body_only: u64,
+    wasm_including_prelude: u64
+}
+
+#[ic_cdk_macros::query]
+pub fn get_perf_result() -> Option<PerfResult> {
+    PERF_RESULT_REF_CELL.with(|perf_result_ref_cell| perf_result_ref_cell.borrow().as_ref().cloned())
+}
+
+fn record_performance(start: u64, end: u64) {
+    PERF_RESULT_REF_CELL.with(|perf_result_ref_cell| {
+        let mut perf_result = perf_result_ref_cell.borrow_mut();
+
+        *perf_result = Some(PerfResult {
+            wasm_body_only: end - start,
+            wasm_including_prelude: ic_cdk::api::call::performance_counter(0)
+        })
+    });
+}
+
+#[derive(candid::CandidType, Clone)]
```

Comment:
I think you haven't pushed your code yet

Reply:
Correct, I shouldn't have made this not a draft... I'm going to blame the lateness of the hour